src/NServiceBus.RavenDB/Outbox/OutboxRecordsCleaner.cs

```diff
@@ -26,38 +30,28 @@ public void RemoveEntriesOlderThan(DateTime dateTime)
 
             try
             {
-                var deletionCommands = new List<ICommandData>();
+                var query = new IndexQuery
+                {
+                    Query = $"Dispatched:true AND DispatchedAt:[* TO {dateTime:o}]"
+                };
 
-                using (var session = DocumentStore.OpenSession())
+                var bulkOpts = new BulkOperationOptions
                 {
-                    var query = session.Query<OutboxRecord, OutboxRecordsIndex>()
-                        .Where(o => o.Dispatched)
-                        .OrderBy(o => o.DispatchedAt);
-
-                    QueryHeaderInformation qhi;
-                    using (var enumerator = session.Advanced.Stream(query, out qhi))
-                    {
-                        while (enumerator.MoveNext())
-                        {
-                            if (enumerator.Current.Document.DispatchedAt >= dateTime)
-                            {
-                                break; // break streaming if we went past the threshold
-                            }
-
-                            deletionCommands.Add(new DeleteCommandData
-                            {
-                                Key = enumerator.Current.Key
-                            });
-                        }
-                    }
-                }
+                    AllowStale = true
+                };
+
+                var operation = documentStore.DatabaseCommands.DeleteByIndex(indexName, query, bulkOpts);
 
-                DocumentStore.DatabaseCommands.Batch(deletionCommands);
+                // This is going to execute multiple "status check" requests to Raven, but
+                // not much we can do about it
+                operation.WaitForCompletion();
```

Comment:
Yeah, otherwise we'll fall into the finally and open up the single-access lock. We're already on a separate thread from message handling activities.